test/test_client_db_hooks.js

```diff
@@ -0,0 +1,63 @@
+import db         from '../src/models/db';
+import * as CONST from './constants';
+import { hookInvoked, resetHooks } from './server';
+
+db().then(models => {
+  describe('Database hooks', () => {
+    beforeEach(() => {
+      resetHooks();
+      return models.sequelize.transaction(transaction => {
+        return Promise.all(Object.keys(CONST.entities).map(name => {
+          return models[name].destroy({ transaction, where: {} });
+        }));
+      });
+    });
+
+    Object.keys(CONST.entities).forEach(entity => {
+      it(entity + ' "create" hooks', done => {
+        models[entity].create(CONST[entity + 'Entity']).then(() => {
+          CONST.dbHooks.slice(0, 2).forEach(hook => {
```

Comment:
nit: Do not make the test dependent on the order of an array that is not within this file, please.

```javascript
[CONST.beforeCreate,
 CONST.afterCreate].forEach(...);
```